src/include/mpir_mem.h

```diff
@@ -84,6 +84,30 @@ extern "C" {
         memcpy((dst), (src), (len));            \
     } while (0)
 
+#if defined(USE_MEMORY_TRACING) && !defined(HAVE_PIP)
```

Comment:
I have been thinking of proposing to remove allocation tracing from MPL, since there are good external tools that provide the same (possibly better) functionality. https://github.com/google/sanitizers/wiki/AddressSanitizer

I will create an issue for discussion.

Reply:
It occurs to me that in addition to tracing, allocation statistics *also* `MPL_malloc` and friends. Perhaps we should document how to disable the preprocessor errors on a per-file basis when including external code which may call `malloc`.

Reply:
BTW, we have this same issue with PMIx. Here is our solution a solution that doesn't require memory tracing. https://github.com/pmodels/mpich/blob/fcb21d8f1c8886f493188863c5d4f7c87704da31/src/include/mpiimpl.h#L112-L116

Any memory allocated by `pip.h` code should be freed with `MPL_external_free`.